crates/bevy_time/src/fixed_timestep.rs

```diff
@@ -106,9 +115,25 @@ pub fn run_fixed_update_schedule(world: &mut World) {
     let mut fixed_time = world.resource_mut::<FixedTime>();
     fixed_time.tick(delta_time);
 
+    let mut ticks_this_frame = 0;
+
+    // Copy this out to appease the borrow checker
+    let maybe_max_ticks = fixed_time.max_ticks_per_frame;
+
     // Run the schedule until we run out of accumulated time
     let _ = world.try_schedule_scope(FixedUpdate, |world, schedule| {
         while world.resource_mut::<FixedTime>().expend().is_ok() {
+            ticks_this_frame +=  1;
+
+            // This is required to avoid entering a death spiral where the fixed update schedule falls behind
+            // and needs to run more and more times per frame to catch up.
+            if let Some(max_ticks) = maybe_max_ticks {
+                if ticks_this_frame > max_ticks {
+                    warn!("The fixed update schedule is falling behind. Consider increasing the period or decreasing the amount of work done in the fixed update schedule.");
```

Comment:
If the game enters the "death spiral", this warning will print out three times per second, constantly. I'm not sure what the policy on warning spam is, so this may be okay.

Also, if the game freezes for a while for any reason (OS freeze for 5 seconds, memory swapping, big blocking loading operation inside game, whatever), these warnings will be printed out for a while (freeze duration seconds * 3 times) until the situation resolves itself. In this case, saying that `update schedule is falling behind` is incorrect, as it is rather catching up to realtime after a freeze.

And if there is a longer freeze, like OS suspend, that lasts for hours or days, then after resuming we'll be stuck doing maximum amount of fixed updates for a very long time as the game catches up to realtime with 20x acceleration but no faster – and the warnings are printed out for the entire duration it is catching up. However, fixing this is out of scope for this pull, so we don't need to care about the behaviour in this case.